php/php.editor/src/org/netbeans/modules/php/editor/verification/IncorrectConstructorPropertyPromotionHintError.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.php.editor.verification;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.netbeans.modules.csl.api.Hint;
+import org.netbeans.modules.csl.api.HintFix;
+import org.netbeans.modules.csl.api.OffsetRange;
+import org.netbeans.modules.csl.spi.support.CancelSupport;
+import org.netbeans.modules.php.editor.CodeUtils;
+import org.netbeans.modules.php.editor.model.FileScope;
+import org.netbeans.modules.php.editor.parser.PHPParseResult;
+import org.netbeans.modules.php.editor.parser.astnodes.ASTNode;
+import org.netbeans.modules.php.editor.parser.astnodes.ArrowFunctionDeclaration;
+import org.netbeans.modules.php.editor.parser.astnodes.Block;
+import org.netbeans.modules.php.editor.parser.astnodes.BodyDeclaration;
+import org.netbeans.modules.php.editor.parser.astnodes.FormalParameter;
+import org.netbeans.modules.php.editor.parser.astnodes.FunctionDeclaration;
+import org.netbeans.modules.php.editor.parser.astnodes.LambdaFunctionDeclaration;
+import org.netbeans.modules.php.editor.parser.astnodes.MethodDeclaration;
+import org.netbeans.modules.php.editor.parser.astnodes.visitors.DefaultVisitor;
+import org.openide.filesystems.FileObject;
+import org.openide.util.NbBundle;
+
+/**
+ * Handle incorrect declaration of Constructor Property Promotion.
+ */
+public class IncorrectConstructorPropertyPromotionHintError extends HintErrorRule {
+
+    private FileObject fileObject;
+
+    @Override
+    @NbBundle.Messages("IncorrectConstructorPropertyPromotionHintError.displayName=Incorrect Declaration of Constructor Property Promotion")
+    public String getDisplayName() {
+        return Bundle.IncorrectConstructorPropertyPromotionHintError_displayName();
+    }
+
+    @Override
+    @NbBundle.Messages({
+        "IncorrectConstructorPropertyPromotionHintError.abstract.desc=Can't declare a promoted property in an abstract constructor.",
+        "IncorrectConstructorPropertyPromotionHintError.non.constructor.desc=Can't declare a promoted property in other than a constructor.",
```

Comment:
@junichi11 It seems that word "function" is missing in this sentence.
"Can't declare a promoted property in other function than a constructor."

Reply:
Will fix it to "can't declare a promoted property in a function/method other than a constructor".